File: ios/AdPie.m
#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_MODULE(AdPie, NSObject)

RCT_EXTERN_METHOD(multiply:(float)a withB:(float)b
                 withResolver:(RCTPromiseResolveBlock)resolve
                 withRejecter:(RCTPromiseRejectBlock)reject)

+ (BOOL)requiresMainQueueSetup
{
  return NO;
}

RCT_EXTERN_METHOD(showAdPieVideoAds:(NSDictionary *)data
                  withResolver:(RCTPromiseResolveBlock)resolve
                  withRejecter:(RCTPromiseRejectBlock)reject)
@end

File: ios/AdPie.swift
@objc(AdPie)
class AdPie: NSObject {

  @objc(multiply:withB:withResolver:withRejecter:)
  func multiply(a: Float, b: Float, resolve:RCTPromiseResolveBlock,reject:RCTPromiseRejectBlock) -> Void {
    resolve(a*b)
  }

  @objc(showAdPieVideoAds:withResolver:withRejecter:)
    func showAdPieVideoAds(_ data: NSDictionary,
     resolver resolve:  RCTPromiseResolveBlock,
      rejecter reject: RCTPromiseRejectBlock) -> Void {
        do {
            let params = data as? [String:AnyObject]
            DispatchQueue.main.async {
                let vc = AdPieVC()
                vc.slotId = params?["slotId"] as! String;
                vc.modalPresentationStyle = .fullScreen
                let controller = RCTPresentedViewController()
                controller?.present(vc, animated: true)
            }
            resolve(true)
        } catch {
            resolve(false)
        }
  }
}
